ddtrace/tracer/tracer.go

```diff
@@ -133,6 +136,12 @@ func newTracer(opts ...StartOption) *tracer {
 	if c.debug {
 		log.SetLevel(log.LevelDebug)
 	}
+
+	statsd, err := statsd.NewBuffered(c.dogstatsdAddr, 40)
+	if err != nil {
+		log.Warn("Runtime and tracer metrics disabled: %v", err)
+		statsd = nil
```

Comment:
IMO this is a big "no" and a bit of a smell. Let's never assign `nil` values to variables in order to define whether something is active or not. I think the only exception where this makes sense is if we have to create a hack for the GC to free up something, but that's rarely the case and questionable when it does happen.

Alternatives are:
* Having a boolean.
* Using the no-op interface, which will also create the opportunity for testing as a positive side effect.